drivers/ieee802154/ieee802154_cc13xx_cc26xx_subg.c

```diff
@@ -368,25 +447,33 @@ ieee802154_cc13xx_cc26xx_subg_get_capabilities(const struct device *dev)
 static int ieee802154_cc13xx_cc26xx_subg_cca(const struct device *dev)
 {
 	struct ieee802154_cc13xx_cc26xx_subg_data *drv_data = dev->data;
+	bool was_rx_on = false;
 	RF_EventMask events;
-	bool was_rx_on;
 	int ret;
 
+	if (k_sem_take(&drv_data->lock, LOCK_TIMEOUT)) {
+		return -EWOULDBLOCK;
+	}
+
+	if (!drv_data->is_up) {
+		ret = -ENETDOWN;
+		goto out;
+	}
+
 	drv_data->cmd_prop_cs.status = IDLE;
-	drv_data->cmd_prop_cs.pNextOp = NULL;
-	drv_data->cmd_prop_cs.condition.rule = COND_NEVER;
 
 	was_rx_on = drv_data->cmd_prop_rx_adv.status == ACTIVE;
-
-	ret = ieee802154_cc13xx_cc26xx_subg_stop(dev);
-	if (ret < 0) {
-		ret = -EIO;
-		goto out;
+	if (was_rx_on) {
+		ret = drv_stop_rx(dev);
+		if (ret) {
+			ret = -EIO;
+			goto out;
+		}
 	}
 
 	events = RF_runCmd(drv_data->rf_handle, (RF_Op *)&drv_data->cmd_prop_cs, RF_PriorityNormal,
 			   NULL, 0);
-	if (events != RF_EventLastCmdDone) {
+	if (events != RF_EventLastCmdDone || drv_data->cmd_prop_cs.status != PROP_DONE_OK) {
```

Comment:
Actually, PROP_DONE_IDLE is returned and the CCA always fails... there's also the switch below that checks the return statuses.

Reply:
OMG. This is really worst quality ever. Embarassing.

Reply:
Fixed.